src/compiler/crystal/macros/interpreter.cr

```diff
@@ -371,15 +371,17 @@ module Crystal
     end
 
     def resolve(node : Path)
+      resolve?(node) || node.raise_undefined_constant(@path_lookup)
+    end
+
+    def resolve?(node : Path)
       if node.names.size == 1 && (match = @free_vars.try &.[node.names.first]?)
         matched_type = match
       else
         matched_type = @path_lookup.lookup_path(node)
       end
 
-      unless matched_type
-        node.raise_undefined_constant(@path_lookup)
-      end
+      return unless matched_type
```

Comment:
That's an interesting stat! I tend to prefer more explicitness given early returning "generates" an union. Then when you get the famous "Nil does not respond to etc." it is easier to see how it can be `Nil`. But I guess my opinion on that is not as pervasive as I would have thought :(